rimage/transform/calibrate_pinhole_extrinsics.go

```diff
@@ -29,7 +29,7 @@ type ExtrinsicCalibrationConfig struct {
 // RunPinholeExtrinsicCalibration will solve the optimization problem to find the rigid pose
 // (translation and rotation) that changes the reference frame from the
 // point of view of the depth camera to the point of the view of the color camera.
-func RunPinholeExtrinsicCalibration(prob *optimize.Problem, logger logging.Logger) (spatialmath.Pose, error) {
+func RunPinholeExtrinsicCalibration(prob *optimize.Problem, logger logging.Logger, deterministic bool) (spatialmath.Pose, error) {
```

Comment:
Because this is a public function I wouldn't want to change the signature just for a test, but I do like the approach! Do you think it'd make sense to only mock out the random function? That is, create a struct that returns a random number `sourceStruct.Float64()` (with a better name haha) that under the hood just calls `rand.Float64()` and others? That way for testing we can just `sourceStruct.Float64 = rand.New(rand.NewSource(0).Float64)` or something more granular. I'm happy to do this as well since its our code.